loopchain/blockchain/votes/v1_0/vote.py

```diff
@@ -157,10 +157,38 @@ def _deserialize(cls, **data) -> Vote:
             height=int(data["blockHeight"], 16)
         )
 
+    def _to_hash(self, origin_data: dict):
+        hash_generator = build_hash_generator(1, "icx_vote")
+        return Hash32(hash_generator.generate_hash(origin_data))
+
+    def _to_origin_data(self):
+        return {
+            "validator": self._voter_id.hex_hx(),
+            "timestamp": hex(self._timestamp),
+            "blockHeight": hex(self._height),
+            "blockHash": Hash32(self._data_id),
+            "commitHash": Hash32(self._commit_id),
+            "stateHash": self._state_hash,
+            "receiptHash": self._receipt_hash,
+            "nextValidatorsHash": self._next_validators_hash,
+            "epoch": self._epoch_num,
+            "round": self._round_num
+        }
+
     def is_none(self) -> bool:
         """Check that the vote is against a block of `data_id`."""
         return self.data_id == self.NoneVote
 
     def is_lazy(self) -> bool:
         """Check that the vote is created by timeout."""
         return self.data_id == self.LazyVote
+
+    def verify(self):
+        origin_data = self._to_origin_data()
+        hash_ = self._to_hash(origin_data)
+        sign_verifier = SignVerifier.from_address(self.voter_id.hex_hx())
+        try:
+            sign_verifier.verify_hash(hash_, self._signature)
+        except Exception as e:
+            raise RuntimeError(f"Invalid vote signature. {self}\n"
+                               f"{e}")
```

Comment:
Multi-line logs make logging much harder (imagine that you parse log lines by using `cat` or `grep` or something).

I suggest to use single line logging.

Reply:
I know what you mean, but single-line logging, which contains multi-line, is bad to read for me when I figure the problem case out. If you suggest to me how the log express better, I'll apply to code.

Reply:
I found LC-805 that contains:
- we decided to leave multiline logs in code lines.

I resolve this review.